### PR TITLE
File download example

### PR DESCRIPTION
**Pull request by bryans99**

### CODE REVIEW

**Review by izzymiller — APPROVED** [review]
Few changes, all just comments / explicatory guidance. Overall LGTM! 

That link clicking mechanic is suboptimal for sure though, maybe something to address in the sandbox at large later. 

thanks for making this example!

File: react/javascript/file-download/src/App.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import React from 'react'
+import { FileDownload } from './FileDownload'
+import { ExtensionProvider2 } from '@looker/extension-sdk-react'
```
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/src/App.js:27]: I think I have it covered in my new documentation for EF.

File: react/javascript/file-download/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Looker Extension File Download Demo (React & JavaScript)
+
+This repository demonstrates a file download Looker extension.
```
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/README.md:3]: Agreed

File: react/javascript/file-download/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Getting Started for Development
+
+1. Clone the Looler extension examples repo.
```
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/README.md:7]: Dammit! My brain correcting for my wonky eye again

File: react/javascript/file-download/manifest.lkml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+project_name: "filedownload"
+
+application: filedownload {
+  label: "File download demo"
+  url: "http://localhost:8080/bundle.js"
+  entitlements: {
+    use_downloads: yes
```
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/manifest.lkml:7]: Done

File: react/javascript/file-download/src/FileDownload.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    // The following is based upon a customer provided example
+    const jsonText = { test: 'test' }
+    const fileName = 'filename.json'
+    const data = new Blob([JSON.stringify(jsonText)], { type: 'text/json' })
+    const jsonURL = window.URL.createObjectURL(data)
+    const link = document.createElement('a')
+    document.body.appendChild(link)
+    link.href = jsonURL
+    link.setAttribute('download', fileName)
+    link.click()
```
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/src/FileDownload.js:54]: This is sample code that I got from a customer to test this out. I thought it was pretty neat. I read up on what it's doing at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a. Not sure I want to explain NON extension framework things. If you think its real helpful I will but I dont want to get into a position where I'm answering JS questions

[the remark this answers is not in the document]
Comment (izzymiller) [reply react/javascript/file-download/src/FileDownload.js:54]: The reason we have to do this is b/c you can't just `window.open(link)` in the ext.framework, right? 

I'm not suggesting we explain the details of the workaround, rather just say "This is to work around the inability to access the parent window" or something. If that is why!

File: react/javascript/file-download/src/App.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryans99) [reply react/javascript/file-download/src/App.js:27]: This is what I wrote in https://docs.google.com/document/d/1aiuE4-w8B0HnRR31FSzkMU5RwrYyCk4-slHcKSjewS4/edit?resourcekey=0-rtWNeqkNIQa0zdKMNeGNJg#

> In an extension, you should pick one of the Looker SDKs, 3.1 or 4.0 and use the ExtensionProvider2 component from the  @looker/extension-sdk-react. If you require both SDKs, continue to use the ExtensionProvider component but you will see an increase in the final bundle size.

Reading it again, its a little light (it says what to do by not really why). I'll make a note to review this part again when Ken comes out with the documentation

**Pull request merged**